packages/mco/components/disaster-recovery/applications/failover-and-relocate-modal/peer-status.tsx

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import { getName } from '@odf/shared/selectors';
+import {
+  StatusIconAndText,
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { TFunction } from 'i18next';
+import { Flex, FlexItem } from '@patternfly/react-core';
+import { UnknownIcon } from '@patternfly/react-icons';
+import { PEER_READINESS } from '../../../../constants';
+import { findPeerStatus } from '../../../../utils';
+import {
+  ACTION_TYPE,
+  FailoverAndRelocateState,
+  FailoverAndRelocateAction,
+  FailoverAndRelocateType,
+} from './reducer';
+
+const initalPeerStatus = (t: TFunction) => ({
+  peerReadiness: {
+    icon: <UnknownIcon />,
+    text: PEER_READINESS(t).UNKNOWN,
+  },
+  lastHealthCheck: {
+    icon: <UnknownIcon />,
+    text: 'Unknown',
+  },
+  lastSnapshotTakenOn: {
+    icon: <UnknownIcon />,
+    text: 'Unknown',
+  },
+});
+
+type StatusProps = {
+  icon: JSX.Element;
+  text: string;
+};
+
+type StatusType = {
+  peerReadiness: StatusProps;
+  lastHealthCheck: StatusProps;
+  lastSnapshotTakenOn: StatusProps;
+};
+
+export const PeerStatus: React.FC<PeerStatusProps> = (props) => {
+  const { t, action, state, dispatch } = props;
+  const [peerStatus, setPeerStatus] = React.useState<StatusType>(
+    initalPeerStatus(t)
+  );
+  const setErrorMessage = React.useCallback(
+    (errorMessage: string) => {
+      dispatch({
+        type: FailoverAndRelocateType.SET_ERROR_MESSAGE,
+        payload: {
+          peerStatusErrorMessage: errorMessage,
+        },
+      });
+    },
+    [dispatch]
+  );
+
+  React.useEffect(() => {
+    if (!!state.selectedSubsGroups.length) {
+      const status = state.acmToDRState?.reduce((arr, acmToDRState) => {
```

Comment:
nits: initial state is an object not array...
```suggestion
      const status = state.acmToDRState?.reduce((obj, acmToDRState) => {
```